packages/interactive/lib/src/executor.dart

```diff
@@ -83,9 +83,12 @@ class Executor {
     log.info('Phase: Evaluate');
     final isolateInfo = await workspaceIsolate.isolateInfo;
     final targetId = isolateInfo.rootLib!.id!;
-    final response = await vm.vmService
+    Response response = await vm.vmService
         .evaluate(workspaceIsolate.isolateId, targetId, _evaluateCode);
-
+    if(response is InstanceRef && response.valueAsString == null) {
+      // InstanceRef.valueAsString only works on primitive values like String, int, double ETC, so for anything else we have to ask the VM to get the toString value
+      response = await vm.vmService.evaluate(workspaceIsolate.isolateId, response.id!, "this.toString()");
+    }
```

Comment:
What about putting this inside _handleEvaluateResponse

Reply:
> What about putting this inside _handleEvaluateResponse

I was doing that at first, but then I saw that it would require complicating the function and making it asynchronous. But now I wonder if we can append a .toString() call directly to the end of the autogenerated method to simplify this. That second response we're sending is probably slowing things down a little.

Reply:
Good idea! Feel free to PR that

> That second response we're sending is probably slowing things down a little.

btw do you feel it in real usage or is it a theoretical thinking?